api/api.go

```diff
@@ -0,0 +1,141 @@
+package api
+
+import (
+	"crypto/tls"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	_ "net/http/pprof"
+
+	"github.com/raintank/met"
+	"github.com/raintank/metrictank/idx"
+	"github.com/raintank/metrictank/mdata"
+	"github.com/raintank/worldping-api/pkg/log"
+	"gopkg.in/macaron.v1"
+)
+
+var LogLevel int
+
+var (
+	getTargetDuration     met.Timer
+	itersToPointsDuration met.Timer
+	// just 1 global timer of request handling time. includes mem/cassandra gets, chunk decode/iters, json building etc
+	// there is such a thing as too many metrics.  we have this, and cassandra timings, that should be enough for realtime profiling
+	reqHandleDuration met.Timer
+	reqSpanBoth       met.Meter
+	reqSpanMem        met.Meter
+)
+
+type Server struct {
+	Addr         string
+	SSL          bool
+	certFile     string
+	keyFile      string
+	Macaron      *macaron.Macaron
+	MetricIndex  idx.MetricIndex
+	MemoryStore  mdata.Metrics
+	BackendStore mdata.Store
+	shutdown     chan struct{}
+}
+
+func (s *Server) BindMetricIndex(i idx.MetricIndex) {
+	s.MetricIndex = i
+}
+func (s *Server) BindMemoryStore(store mdata.Metrics) {
+	s.MemoryStore = store
+}
+func (s *Server) BindBackendStore(store mdata.Store) {
+	s.BackendStore = store
+}
+
+func NewServer(addr string, stats met.Backend) (*Server, error) {
+	//validate the addr
+	_, err := net.ResolveTCPAddr("tcp", addr)
+	if err != nil {
+		return nil, err
+	}
+
+	reqSpanMem = stats.NewMeter("requests_span.mem", 0)
+	reqSpanBoth = stats.NewMeter("requests_span.mem_and_cassandra", 0)
+	getTargetDuration = stats.NewTimer("get_target_duration", 0)
+	itersToPointsDuration = stats.NewTimer("iters_to_points_duration", 0)
+	reqHandleDuration = stats.NewTimer("request_handle_duration", 0)
+
+	m := macaron.Classic()
```

Comment:
this will also make it serve static files which we don't have. why not change this to:
```
	m := macaron.New()
	m.Use(macaron.Logger())
	m.Use(macaron.Recovery())
```